renameat2 test by meson failing with glibc 2.28-4, linux-headers 4.18.5 on Arch

Building e4a3c5efc8f11e0e99f8cc97bd417665d92b40a9 on current Arch Linux, with glibc 2.28-4, `meson.build` detects `renameat2` as unavailable: 

```
Checking for function "renameat2" : NO
```

...followed by, later:

```
In file included from ../src/caformat-util.h:9,
                 from ../src/caremote.c:10:
../src/util.h:532:19: error: static declaration of ‘renameat2’ follows non-static declaration
 static inline int renameat2(int oldfd, const char *oldname, int newfd, const char *newname, unsigned flags) {
                   ^~~~~~~~~
In file included from ../src/util.h:12,
                 from ../src/caformat-util.h:9,
                 from ../src/caremote.c:10:
/usr/include/stdio.h:164:12: note: previous declaration of ‘renameat2’ was here
 extern int renameat2 (int __oldfd, const char *__old, int __newfd,
            ^~~~~~~~~
ninja: build stopped: subcommand failed.
```

---

Oddly, though (given that `/usr/include/stdio.h` *does* define `renameat2` (when `__USE_GNU` is set), digging through `meson-log.txt`, it's genuinely not there during the test -- something which remains true when editing `meson.build` to add `#define __USE_GNU 1` to the invocation:

```
Command line:  cc /tmp/tmp04xnec4t/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmp04xnec4t/output.exe -march=x86 -64 -mtune=generic -O2 -fstack-protector-strong -fno-plt -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -O0 -std=gnu99

Code:
 #include <stdio.h>
#include <limits.h>

        #if defined __stub_renameat2 || defined __stub___renameat2
        fail fail fail this function is not going to work
        #endif

int main() {
            void *a = (void*) &renameat2;
            long b = (long) a;
            return (int) b;
        }
Compiler stdout:

Compiler stderr:
 /tmp/tmp04xnec4t/testfile.c: In function 'main':
/tmp/tmp04xnec4t/testfile.c:9:32: error: 'renameat2' undeclared (first use in this function); did you mean 'renameat'?
             void *a = (void*) &renameat2;
                                ^~~~~~~~~
                                renameat
```

--- a/meson.build
+++ b/meson.build
@@ -78,7 +78,7 @@ foreach ident : [
         ['copy_file_range',   '''#include <sys/syscall.h>
                                  #include <unistd.h>'''],
 ]
-        have = cc.has_function(ident[0], prefix : ident[1])
+        have = cc.has_function(ident[0], args : '-D_GNU_SOURCE', prefix : ident[1])
         conf.set10('HAVE_' + ident[0].to_upper(), have)
 endforeach
 

